### ISSUE TITLE
autorotate random access (VipsJpeg: out of order read)  issue with vips 8.10

### ISSUE THREAD

**Issue by chregu** [libvips/php-vips#105]
I have this strange issue, but just with the PHP extension, not with the vips commandline (couldn't recreate the issue there)

The following worked fine with 8.9

```
$vips = VipsImage::newFromFile("foo.jpg", ["autorotate" => true]);
$vips->jpegsave("result.jpg");
```
with the following jpg
![foo](https://user-images.githubusercontent.com/47106/92108660-04fb6000-ede8-11ea-8bc3-c89991221da9.jpg)

in 8.10 I get a several 
```
(banana:278): VIPS-WARNING **: error in tile 0 x 208
VipsJpeg: out of order read at line 3024
```

But if I set the "access" to "sequential", like this

```
$vips = VipsImage::newFromFile("foo.jpg", ["autorotate" => true, "access" => 'sequential']);
$vips->jpegsave("result.jpg");
```

it works fine. I'd expect it the other way round with that error, that it should have access = random (which is the default, AFAIK).

Any idea, what could be wrong here? Just setting "access" to "sequential" everywhere might also not be the right choice for me.

**Comment by jcupitt** [comment]
Hi @chregu,

You're right, the autorotate option to jpeg load seems to be broken. I'll investigate.

In the meantime, you can use the separate `autorot` method, eg.:

```php
$vips = VipsImage::newFromFile("foo.jpg");
$vips = $vips->autorot();
$vips->jpegsave("result.jpg");
```

**Comment by jcupitt** [comment]
OK, fixed in master and the fix should be in 8.10.1. Thanks for the report!

**Comment by chregu** [comment]
Thanks a lot. seems indeed to be fixed now (tested on 8.10 branch)
